hathor/builder/builder.py

```diff
@@ -131,7 +132,7 @@ def __init__(self) -> None:
         self._tx_storage_cache_capacity: Optional[int] = None
 
         self._indexes_manager: Optional[IndexesManager] = None
-        self._tx_storage: Optional[TransactionStorage] = None
+        self._tx_storage: Optional[BaseTransactionStorage] = None
```

Comment:
`BaseTransactionStorage` is not supposed to be used outside of `hathor.transaction.storage.transaction_storage`, it is meant as a storage that provides a basic implementation of some of the interfaces of `TransactionStorage`.

I wonder why this change was needed, but I'd rather fix `TransactionStorage` instead, maybe it is missing an abstract method?

This comment applies to many of the changes in the PR, I'll leave this only here, but I'll review the other places before closing this thread.

Reply:
> maybe it is missing an abstract method?

Yes, that was the case, it was just missing a property. I fixed it and changed all back to `TransactionStorage` in https://github.com/HathorNetwork/hathor-core/pull/973/commits/1f362afe88d3584877041d87b543837f4ae9440c